firedrake/slope_limiter/limiter.py

```diff
@@ -0,0 +1,121 @@
+from firedrake import *
+from abc import *
+
+class _Limiter(object):
+
+    __metaclass__ = ABCMeta
+
+    @abstractmethod
+    def apply(self, field):
+        return NotImplemented
```

Comment:
The `@abstractmethod` property requires that subclasses implement this method themselves, so no need to raise `NotImplemented`.  You can just write `pass`.  For future reference, you should `raise NotImplementedError` (not `NotImplemented`).

Again, here and below, docstring describing what this method does.